.github/workflows/ci.yml

```diff
@@ -79,3 +105,45 @@ jobs:
         export PATH=$PATH:/c/Users/runneradmin/AppData/Roaming/local/bin;
         bats ./tests/golden/**
       shell: bash
+
+  xrefcheck-build-and-test-nix:
+    runs-on: [self-hosted, nix]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: build
+        run: nix build -L .#xrefcheck
+
+      - name: test
+        run: nix build -L .#checks.x86_64-linux.test
+
+      - name: Golden tests (bats)
+        run: nix shell .#bats .#diffutils .#xrefcheck -c bash -c "cd tests/golden/ && bats ./**"
+
+      - name: windows cross-compilation
+        run: nix build -L .#xrefcheck-windows
+
+      - name: Upload windows executable
+        uses: actions/upload-artifact@v3
+        with:
+          name: xrefcheck-windows
+          path: |
+            result/bin/*
+            result/lib/*
+
+      - name: docker-image
+        run: nix build -L .#docker-image
+
+      - name: static binary
+        run: nix build -L .#xrefcheck-static
+
+      - name: Upload static binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: xrefcheck-static
+          path: result/bin/xrefcheck
+
+      - name: Xrefcheck itself
+        run: nix run . -L
```

Comment:
We had
```
    retry:
      automatic:
        limit: 2
```
for it

Reply:
Ugh, nick-fields/retry@v2.8.3 doesn't work well with nix:(

Reply:
Does it mean there is no way for us to configure some step to automatically retry N times?

Reply:
I think there is a way, but this will require some additional research to fix but I didn't want to spend too much time on this right now:(